### PR TITLE
 Add support for HTML, CSS and Javascript in LocalCommandLineCodeExecutor with Mapping executor/saver #2303 

### PR DESCRIPTION
**Pull request by asandez1**
### Why are these changes needed?

Many web application development use cases require support for simply saving the static content like HTML, CSS and Javascript.

This PR adds support for those languages in LocalCommandLineCodeExecutor.

It also add option in LocalCommandLineCodeExecutor for only saving the code files not executing them. This is useful when the web application code starts a server and never exits.

-EXECUTION_POLICIES mapping of language to whether it should be executed or just saved

Related issue number

Closes https://github.com/microsoft/autogen/issues/2379 https://github.com/microsoft/autogen/issues/2383

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ x] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by asandez1** [comment]
@microsoft-github-policy-service agree

File: autogen/coding/local_commandline_code_executor.py [hunk 1]
```diff
@@ -91,6 +108,8 @@ def __init__(
         else:
             self._setup_functions_complete = True
 
+        self.execution_policies = execution_policies if execution_policies is not None else self.EXECUTION_POLICIES
```
Comment (ekzhu) [inline autogen/coding/local_commandline_code_executor.py:111]: Should create a copy of the default policy, merge and override the default if the user provides a policy dictionary, as the user-provided one can be in-complete.

File: autogen/coding/local_commandline_code_executor.py [hunk 2]
```diff
@@ -28,7 +28,31 @@
...
+        "sh",
+        "pwsh",
+        "powershell",
+        "ps1",
+        "python",
+        "javascript",
+        "html",
+        "css",
+    ]
+    EXECUTION_POLICIES: ClassVar[Dict[str, bool]] = {
```
[the remark this answers is not in the document]
Comment (ekzhu) [reply autogen/coding/local_commandline_code_executor.py:43]: Should be called "DEFAULT_EXECUTION_POLICY"

File: autogen/coding/local_commandline_code_executor.py [hunk 3]
```diff
@@ -239,15 +249,20 @@ def _execute_code_dont_check_setup(self, code_blocks: List[CodeBlock]) -> Comman
                 # create a file with an automatically generated name
                 code_hash = md5(code.encode()).hexdigest()
                 filename = f"tmp_code_{code_hash}.{'py' if lang.startswith('python') else lang}"
-
             written_file = (self._work_dir / filename).resolve()
             with written_file.open("w", encoding="utf-8") as f:
                 f.write(code)
             file_names.append(written_file)
+            print(f"Language: {lang}, Execute: {execute_code}")
```
Comment (ekzhu) [inline autogen/coding/local_commandline_code_executor.py:256]: Let's not print here.

File: autogen/coding/local_commandline_code_executor.py [hunk 4]
```diff
@@ -271,70 +283,3 @@ def _execute_code_dont_check_setup(self, code_blocks: List[CodeBlock]) -> Comman
     def restart(self) -> None:
         """(Experimental) Restart the code executor."""
         warnings.warn("Restarting local command line code executor is not supported. No action is taken.")
-
```
Comment (ekzhu) [inline autogen/coding/local_commandline_code_executor.py:274]: Why should we remove this? 

cc @jackgerrits

**Review by ekzhu — CHANGES_REQUESTED** [review]
Thanks for the PR.

File: autogen/coding/local_commandline_code_executor.py (hunk 1 above)
Comment (asandez1) [reply autogen/coding/local_commandline_code_executor.py]: The  class will safely use a mix of default and user-specified policies without risking unexpected side effects from shared mutable objects, and it ensures that all supported languages have a defined policy. Thanks for the comment.

File: autogen/coding/local_commandline_code_executor.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (asandez1) [reply autogen/coding/local_commandline_code_executor.py]: I renamed the policy and allow the user to provide a new policy.

File: autogen/coding/local_commandline_code_executor.py [hunk 5]
```diff
@@ -43,29 +67,22 @@ def __init__(
...
-        the working directory, and a unique file is generated and saved in the
-        working directory for each code block.
-        The code blocks are executed in the order they are received.
-        Command line code is sanitized using regular expression match against a list of dangerous commands in order to prevent self-destructive
-        commands from being executed which may potentially affect the users environment.
-        Currently the only supported languages is Python and shell scripts.
-        For Python code, use the language "python" for the code block.
-        For shell scripts, use the language "bash", "shell", or "sh" for the code
-        block.
+        """
```
[the remark this answers is not in the document]
Comment (asandez1) [reply autogen/coding/local_commandline_code_executor.py]: Added!

File: autogen/coding/local_commandline_code_executor.py (hunk 3 above)
Comment (asandez1) [reply autogen/coding/local_commandline_code_executor.py]: good catch

File: autogen/coding/local_commandline_code_executor.py (hunk 4 above)
Comment (asandez1) [reply autogen/coding/local_commandline_code_executor.py]: Added

**Review by ekzhu — APPROVED** [review]

**Pull request merged**